helix-view/src/document.rs

```diff
@@ -634,6 +654,15 @@ impl Document {
         // and error out when document is saved
         self.path = path;
 
+        if self.text.len_bytes() > BIG_FILE_THRESHOLD {
+            self.enable_syntax = false;
```

Comment:
There should be a message saying that syntax is disabled or maybe some signs, or otherwise the user might find it surprising.